include/common_package/commands.hpp

```diff
@@ -0,0 +1,214 @@
+            this->min_val = min_val;
+            this->max_val = max_val;
+        }
+    };
+
+    /**
+     * @brief The CommandDefinitions class provides a set of static functions to define and check command keys for different types.
+     *
+     * The CommandDefinitions class contains static functions to define the keys for different commands and check if a JSON string matches the definition.
+     * It provides a map of command keys and their corresponding definitions, as well as a function to check a JSON string against the defined command keys.
+     */
+    class CommandDefinitions
+    {
+    public:
+        /**
+         * @brief Parses and checks a string formatted JSON against a given definition
+         *
+         * - Checks that no undefined keys are in the JSON
+         * - Checks that all required keys exist
+         * - Checks that all values have the correct type
+         *
+         * @note The function can only be used on shallow JSONs. Arrays or encapsulations are not supported and will result in an exception!
+         *
+         * @param json_str String formatted json that should be parsed and checked
+         * @param definition Definition of what keys shall be included and what type they can have
+         * @throws std::runtime_error with an error message why the check failed
+         */
+        static nlohmann::json parse_check_json(const std::string &json_str, const std::map<std::string, JsonKeyDefinition> definition);
+
+        /**
+         * @brief Checks a JSON object against a given definition
+         *
+         * Only use this version if you already have a json object.
+         * If you only have a string, use the other overload instead to do a proper parsing.
+         *
+         * - Checks that no undefined keys are in the JSON
+         * - Checks that all required keys exist
+         * - Checks that all values have the correct type
+         *
+         * @note The function can only be used on shallow JSONs. Arrays or encapsulations are not supported and will result in an exception!
+         *
+         * @param json_obj Json object that should be checked
+         * @param definition Definition of what keys shall be included and what type they can have
+         * @throws std::runtime_error with an error message why the check failed
+         */
+        static nlohmann::json parse_check_json(const nlohmann::json &json_obj, const std::map<std::string, JsonKeyDefinition> definition);
+
+        /**
+         * Returns a map containing the definition of waypoint command keys.
+         *
+         * The keys in the map are of type std::string, representing the names of the command keys.
+         * The values in the map are of type JsonKeyDefinition, representing the definition of each command key.
+         *
+         * @return A map containing the definition of waypoint command keys.
+         */
+        static std::map<std::string, JsonKeyDefinition> get_waypoint_command_definition()
```

Comment:
Wieso die map nicht statisch erstellen und dann eine const ref darauf zurück geben?
```suggestion
        static const std::map<const std::string, const JsonKeyDefinition>& get_waypoint_command_definition()
```

Reply:
Bis auf die Referenz alles hinzugefügt. Referenz auf eine lokale Variable funktioniert nicht.